graylog2-server/src/test/java/org/graylog/testing/UserServiceExtension.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.testing;
+
+import org.graylog.testing.mongodb.MongoDBExtension;
+import org.graylog.testing.mongodb.MongoDBTestService;
+import org.graylog2.shared.users.UserService;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+public class UserServiceExtension implements ParameterResolver {
+    @Override
+    public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+        return Objects.equals(UserService.class, parameterContext.getParameter().getType());
+    }
+
+    @Override
+    public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+        final Class<?> parameterType = parameterContext.getParameter().getType();
+
+        if (UserService.class.equals(parameterType)) {
```

Comment:
Should we only inject `TestUserService` instances? That would make it clear that the user of the extension gets a `TestUserService` and not a fully featured `UserServiceImpl`. (we might want to rename the extension to `TestUserServiceExtension` in that case, too)

```suggestion
        if (TeamUserService.class.equals(parameterType)) {
```

Reply:
Makes sense :+1: